src/installer.ts

```diff
@@ -167,6 +167,10 @@ function normalizeVersion(version: string): string {
 }
 
 async function determineVersion(version: string): Promise<string> {
+  if (version == 'latest' || version == 'tip') {
```

Comment:
I left this comment on the commit directly, but I would strongly advise against checking tip here. If it doesn't actually install tip, then it's going to be misleading when someone ends up with the latest stable build instead.

#21 is about actually building tip, like `gimme` does on Travis. Needing an extra step in the workflow anyway makes this check not worthwhile, since the step would have worked without this change (see https://github.com/actions/setup-go/issues/21#issuecomment-565704236).

Reply:
I left the comment on the commit too. https://github.com/tie/setup-go/commit/ff6bc1640078e249753d4f5ff3bb74eb647df2a6#commitcomment-36429256

Why would the following (https://github.com/actions/setup-go/issues/21#issuecomment-565704236) work when `tip` is in a build matrix? Shouldn’t it be ignored because of the `matrix.go != 'tip'`?

```yaml
- name: Install Go
  if: matrix.go != 'tip'
  uses: actions/setup-go@v1
  with:
    go-version: ${{ matrix.go }}
```

Reply:
Matrixes are separate from actions. If it's not tip, `setup-go` runs. If it's tip, then `setup-go` does not run, and `Go` is installed via `gotip` using the builder's preinstalled Go version: https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners

It doesn't matter what version you start with to build tip; Go has no problem building itself from versions even as old as Go 1.4 (the last version written in C).

I plan to update my comment to make it more efficient; installing `gotip` is just overhead that's wasting time.

Reply:
Ah, sorry, didn’t read the whole comment.

>No need to run setup-go to bootstrap this stage, since the builders come with a Go version already installed.

Reply:
Of course, removing that assumption would be preferred; I don't know if self-hosted builders can make all of the same assumptions as Github-run ones that bundle a bunch of stuff. For tip support in setup-go, I'd probably remove that assumption (and it won't be too slow, thanks to the caching that's happening when setup-go runs).

Reply:
In this case there is really no need for `tip` alias.

Reply:
I’ll look into adding built-in `tip` support in a separate PR later.